blockchain/chain.go

```diff
@@ -1089,51 +1102,52 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 			return err
 		}
 
-		// Update the database and chain state.
+		// Update the database and chain state. The cache will be flushed
+		// here before the utxoview modifications happen to the database.
 		err = b.disconnectBlock(n, block, view)
 		if err != nil {
 			return err
 		}
+
+		// Flush the cache again after the disconnect block because the
+		// utxo cache briefly becomes accessible due to the chainLock
+		// being unlocked for the notification callback.
+		//
+		// This is needed as we cannot leave the utxo set in an inconsistent
+		// state during block disconnects.
+		err = b.db.Update(func(dbTx database.Tx) error {
```

Comment:
Is this needed given that in `disconnectBlock`, we'll now (as of this commit) flush the cache in the same db txns that we update the rest of the indexes in?

Reply:
Yeah this isn't needed if we get rid of the option to fetch from the database in `fetchInputUtxos`

Reply:
Addressed in the latest push